Seastar cannot be compiled with gcc 13.1 and fmt lib version 10

```
In static member function ‘static constexpr _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = unsigned int; _Up = unsigned int; bool _IsMove = false]’,
    inlined from ‘constexpr _OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned int*; _OI = unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘constexpr _OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned int*; _OI = unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘constexpr _OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned int*; _OI = unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘constexpr _OI std::copy(_II, _II, _OI) [with _II = unsigned int*; _OI = unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘static _ForwardIterator std::__uninitialized_copy<true>::__uninit_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = unsigned int*; _ForwardIterator = unsigned int*]’ at /usr/include/c++/13/bits/stl_uninitialized.h:147:27,
    inlined from ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = unsigned int*; _ForwardIterator = unsigned int*]’ at /usr/include/c++/13/bits/stl_uninitialized.h:185:15,
    inlined from ‘constexpr void fmt::v9::basic_memory_buffer<T, SIZE, Allocator>::grow(size_t) [with T = unsigned int; long unsigned int SIZE = 32; Allocator = std::allocator<unsigned int>]’ at /usr/include/fmt/format.h:925:26,
    inlined from ‘constexpr void fmt::v9::detail::buffer<T>::try_reserve(size_t) [with T = unsigned int]’ at /usr/include/fmt/core.h:928:39,
    inlined from ‘constexpr void fmt::v9::detail::buffer<T>::try_reserve(size_t) [with T = unsigned int]’ at /usr/include/fmt/core.h:927:24,
    inlined from ‘constexpr void fmt::v9::detail::buffer<T>::try_resize(size_t) [with T = unsigned int]’ at /usr/include/fmt/core.h:919:16,
    inlined from ‘constexpr void fmt::v9::basic_memory_buffer<T, SIZE, Allocator>::resize(size_t) [with T = unsigned int; long unsigned int SIZE = 32; Allocator = std::allocator<unsigned int>]’ at /usr/include/fmt/format.h:897:63,
    inlined from ‘constexpr void fmt::v9::detail::bigint::assign(UInt) [with UInt = long unsigned int; typename std::enable_if<(std::is_same<UInt, long unsigned int>::value || std::is_same<UInt, __int128 unsigned>::value), int>::type <anonymous> = 0]’ at /usr/include/fmt/format.h:2792:19,
    inlined from ‘constexpr void fmt::v9::detail::bigint::operator=(Int) [with Int = int]’ at /usr/include/fmt/format.h:2813:11,
    inlined from ‘constexpr void fmt::v9::detail::bigint::assign_pow10(int)’ at /usr/include/fmt/format.h:2886:32:
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ writing between 5 and 9223372036854775807 bytes into a region of size 4 overflows the destination [-Werror=stringop-overflow=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/13/x86_64-redhat-linux/bits/c++allocator.h:33,
                 from /usr/include/c++/13/bits/allocator.h:46,
                 from /usr/include/c++/13/string:43:

```

Can be reduced to https://godbolt.org/z/osqjxs8Wq

The issue seems to be worked around in fmt's master

Relevant discussion https://github.com/fmtlib/fmt/issues/3382

Reading quickly through it, they seem to have traced it to changes in implicit move rules, described in https://gcc.gnu.org/gcc-13/porting_to.html

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -767,6 +767,14 @@ set (Seastar_PRIVATE_CXX_FLAGS
   -Wimplicit-fallthrough
   -Wdeprecated
   -Wno-error=deprecated)
+
+include (CheckGcc107852)
+if (NOT Cxx_Compiler_BZ107852_Free)
+  list (APPEND Seastar_PRIVATE_CXX_FLAGS
+    -Wno-error=stringop-overflow
+    -Wno-error=array-bounds)
+endif ()
+
 if (NOT BUILD_SHARED_LIBS)
   list (APPEND Seastar_PRIVATE_CXX_FLAGS -fvisibility=hidden)
 endif ()

--- a/cmake/CheckGcc107852.cmake
+++ b/cmake/CheckGcc107852.cmake
@@ -1,0 +1,24 @@
+include (CheckCXXSourceCompiles)
+include (CMakePushCheckState)
+
+cmake_push_check_state (RESET)
+
+# these options are included by -Wall, which is in turn included by
+# Seastar_PRIVATE_CXX_FLAGS, which is not applied to CMAKE_CXX_FLAGS, so
+# let's apply them explicitly.
+set (CMAKE_REQUIRED_FLAGS "-Werror=stringop-overflow -Werror=array-bound")
+set (CMAKE_REQUIRED_LIBRARIES fmt::fmt)
+
+# see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107852
+check_cxx_source_compiles ("
+#include <fmt/ranges.h>
+
+int main() {
+    float grades[] = {3.14};
+    fmt::print(\"{}\", grades);
+}"
+  Cxx_Compiler_BZ107852_Free
+  FAIL_REGEX "is out of the bounds")
+
+cmake_pop_check_state ()
+

